lib/travis/services/remove_log.rb

```diff
@@ -33,21 +33,31 @@ def run
           message << "\n\n#{params[:reason]}"
         end
 
-        log.clear!
-        log.update_attributes!(
-          :content => nil,
-          :aggregated_at => nil,
-          :removed_at => removed_at,
-          :removed_by => current_user
-        )
-        log.parts.create(content: message, number: 1, final: true)
+
+        if Travis.config.logs_api.enabled?
+          log.clear!(current_user)
+        else
+          log.clear!
+          log.update_attributes!(
+            :content => nil,
+            :aggregated_at => nil,
+            :removed_at => removed_at,
+            :removed_by => current_user
+          )
+          log.parts.create(content: message, number: 1, final: true)
+        end
+
         log
       end
 
       instrument :run
 
       def log
-        @log ||= job.log
+        @log ||= if Travis.config.logs_api.enabled?
```

Comment:
If you take my other suggestions about the log method on the 2 job models you don't need to modify this logic.